GVRf/Framework/src/org/gearvrf/scene_objects/GVRViewSceneObject.java

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.scene_objects;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRDrawFrameListener;
+import org.gearvrf.GVRExternalTexture;
+import org.gearvrf.GVRMaterial;
+import org.gearvrf.GVRMaterial.GVRShaderType;
+import org.gearvrf.GVRMesh;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRTexture;
+import org.gearvrf.scene_objects.view.GVRView;
+
+import android.graphics.Canvas;
+import android.graphics.SurfaceTexture;
+import android.view.Surface;
+
+/**
+ * This class represents a {@linkplain GVRSceneObject Scene object} that shows a {@link View}
+ * into the scene with an arbitrarily complex geometry.
+ * See {@link GVRView}
+ */
+public class GVRViewSceneObject extends GVRSceneObject implements
+        GVRDrawFrameListener {
+
+    private final Surface mSurface;
+    private final SurfaceTexture mSurfaceTexture;
+
+    /**
+     * Shows any view into the {@linkplain GVRViewSceneObject scene object} with an arbitrarily complex
+     * geometry.
+     * 
+     * @param gvrContext current {@link GVRContext}
+     * @param gvrView The {@link GVRView} to be shown.
+     * @param mesh a {@link GVRMesh} - see
+     *            {@link GVRContext#loadMesh(org.gearvrf.GVRAndroidResource)} and
+     *            {@link GVRContext#createQuad(float, float)}
+     */
+    public GVRViewSceneObject(GVRContext gvrContext, GVRView gvrView, GVRMesh mesh) {
+        super(gvrContext, mesh);
+
+        gvrContext.registerDrawFrameListener(this);
+        GVRTexture texture = new GVRExternalTexture(gvrContext);
+
+        // TODO: Shader type maybe defined by some GVRView.getShaderType()
+        // according to view type
+        GVRMaterial material = new GVRMaterial(gvrContext, GVRShaderType.OES.ID);
+        material.setMainTexture(texture);
+        getRenderData().setMaterial(material);
+
+        mSurfaceTexture = new SurfaceTexture(texture.getId());
+        mSurface = new Surface(mSurfaceTexture);
+        mSurfaceTexture.setDefaultBufferSize(gvrView.getView().getWidth(),
```

Comment:
Hi @thomasflynn,
Maybe I haven't understood what you was meaning ...
But it is just breaking the previous line, because of that it has 8 spaces.